src/traversal/printRelayOSSQuery.js

```diff
@@ -301,6 +301,14 @@ function createVariable(
   type: string,
   printerState: PrinterState
 ): string {
+  const existingIds = Object.keys(printerState.variableMap);
+  for (let i = 0; i < existingIds.length; i++) {
+    const existingId = existingIds[i];
+    const existingVariable = printerState.variableMap[existingId];
+    if (existingVariable.type === type && Object.is(existingVariable.value, value)) {
+      return existingId;
+    }
+  }
```

Comment:
This will make each call to `printRelayQuery` O(n^2) with the number of variables in the query. At the moment we inline most argument values instead of creating variables, but if we ever changed that the use of a `for` loop here could be prohibitively expensive. 

Instead, how about changing the printer state's `variableMap` to a `Map` (to allow objects as keys since `value` is mixed), and changing it from `{[variableID: string]: {type,value}}` to `Map<value, {type, variableID}>`. Then the logic here can be roughly:

```javascript
const {variableMap} = printerState;
if (variableMap.has(value)) {
  return variableMap.get(value).variableID;
} else {
  const variableID = name + '_' + base62(printerState.variableCount++);
  variableMap.set(value, {type, variableID});
  return variableID;
}
```

The final printed variables can be created by iterating over `variableMap` to produce an object of variableID -> value.